examples/automatic_speech_recognition/automatic_speech_recognition/seed_test_suite.py

```diff
@@ -0,0 +1,357 @@
+        reset=True,
+    )
+    print(f"created test suite {test_suite.name} v{test_suite.version}")
+
+
+def seed_test_suite_zero_crossing_rate(
+    test_suite_name: str,
+    complete_test_case: TestCase,
+) -> TestSuite:
+    test_case_name_to_decision_logic_map = {
+        "1st quartile": lambda x: x < 1771,
+        "2nd quartile": lambda x: 1771 <= x < 2292,
+        "3rd quartile": lambda x: 2292 <= x < 2838,
+        "4th quartile": lambda x: 2838 <= x,
+    }
+
+    test_cases = []
+    for name, fn in test_case_name_to_decision_logic_map.items():
+        ts_list = [
+            (ts, gt) for ts, gt in complete_test_case.iter_test_samples() if fn(ts.metadata["zero_crossing_rate"])
+        ]
+
+        new_ts = TestCase(
+            f"zero crossing rate :: {name} :: {DATASET}",
+            test_samples=ts_list,
+            reset=True,
+        )
+        test_cases.append(new_ts)
+
+    test_suite = TestSuite(
+        test_suite_name,
+        test_cases=[complete_test_case, *test_cases],
+        reset=True,
+    )
+    print(f"created test suite {test_suite.name} v{test_suite.version}")
+
+
+def seed_test_suite_tempo(
+    test_suite_name: str,
+    complete_test_case: TestCase,
+) -> TestSuite:
+    test_case_name_to_decision_logic_map = {
+        "slower": lambda x: x < 2.5,
+        "medium": lambda x: 2.5 <= x < 3.1,
+        "faster": lambda x: 3.1 <= x,
+    }
+
+    test_cases = []
+    for name, fn in test_case_name_to_decision_logic_map.items():
+        ts_list = [(ts, gt) for ts, gt in complete_test_case.iter_test_samples() if fn(ts.metadata["tempo"])]
+
+        new_ts = TestCase(
+            f"tempo (words per second) :: {name} :: {DATASET}",
+            test_samples=ts_list,
+            reset=True,
+        )
+        test_cases.append(new_ts)
+
+    test_suite = TestSuite(
+        test_suite_name,
+        test_cases=[complete_test_case, *test_cases],
+        reset=True,
+    )
+    print(f"created test suite {test_suite.name} v{test_suite.version}")
+
+
+def seed_complete_test_case(args: Namespace) -> TestCase:
+    df = pd.read_csv(args.dataset_csv)
+    df = df.where(pd.notnull(df), None)  # read missing cells as None
+    df.columns = df.columns.str.replace(r"(\s|\.)+", "_", regex=True)  # sanitize column names to use underscores
+
+    required_columns = {
+        "id",
+        "locator",
+        "text",
+        "inference_whisper_default",
+        "inference_wav2vec-base-960h",
+        "speaker_sex",
+        "duration_seconds",
+        "max_pitch",
+        "energy",
+        "zero_crossing_rate",
+        "word_count",
+        "longest_word_len",
+        "tempo",
+    }
+    assert all(required_column in set(df.columns) for required_column in required_columns)
+
+    test_samples = []
+    for record in tqdm(df.itertuples(index=False), total=len(df)):
+        test_sample = TestSample(
+            locator=f"s3://{BUCKET}/{DATASET}/{record.locator}",
+            metadata={
```

Comment:
Is there a reason for not doing this: `metadata={f: getattr(record, f) for f in set(record._fields) - required_columns - optional_columns},`?